certification/internal/bindings/types.go

```diff
@@ -0,0 +1,54 @@
+package bindings
+
+import (
+	"context"
+
+	"github.com/docker/docker/client"
+)
+
+// PodmanClient is a client used to interact with
+// containers and images via the with podman socket
+type PodmanClient struct {
+	context.Context
+}
+
+// DockerSocket is a client used to interact with
+// containers and images via the with podman socket
+type DockerClient struct {
+	*client.Client
+	context.Context
+}
+
+// Container struct provides a way to standardize the results returned
+// by the methods run by the podman client and docker client methods
+type Container struct {
+	ID     string
+	Names  []string
+	Labels map[string]string
+}
+
+// Image struct provides a way to standardize the results returned
+// by the methods run by the podman client and docker client methods
+type Image struct {
+	ID          string
+	Labels      map[string]string
+	RepoTags    []string
+	RepoDigests []string
+}
+
+// The ContainerTool interface provides a unified way to interact with
+// both the podman and docker methods below
+type ContainerTool interface {
+	// PullImage takes an image name and pulls the image to the local cache
```

Comment:
Most of these methods should have a return type of a struct specific to that operation (PullImageReport, ListImageReport, etc.). It would "future proof" the return type a bit, so that as more structured data is returned, existing code wouldn't necessarily need to change.

Reply:
roger that